pallets/liquid-staking/src/lib.rs

```diff
@@ -310,8 +323,19 @@ pub mod pallet {
         ) -> DispatchResultWithPostInfo {
             T::WithdrawOrigin::ensure_origin(origin)?;
 
+            let reserve = Self::reserve_factor().mul_floor(amount);
+            TotalReserve::<T>::try_mutate(|b| -> DispatchResult {
+                *b = b.checked_add(reserve).ok_or(ArithmeticError::Overflow)?;
+                Ok(())
+            })?;
+
+            let left_amount = amount
+                .checked_sub(reserve)
+                .ok_or(ArithmeticError::Overflow)?;
             TotalStakingAsset::<T>::try_mutate(|b| -> DispatchResult {
-                *b = b.checked_add(amount).ok_or(ArithmeticError::Overflow)?;
+                *b = b
```

Comment:
With this change, `TotalStakingAsset` will not be the actual value, since portion of the reward is only recorded in `TotalReserve`. 
I don' know if this going to be an issue in the long term, but let's think about how to improve it.

Reply:
yes, got it.